src/app/wallets/ln-send-payment.ts

```diff
@@ -481,9 +491,18 @@ const executePaymentViaLn = async ({
           milliSatsAmount: toMilliSatsFromNumber(amount * 1000),
           maxFee,
         })
+
+    const payment = await lndService.lookupPayment({ pubkey, paymentHash })
+    if (payment instanceof Error) return payment
+
+    payment.paymentRequest = payment.paymentRequest || paymentRequest
```

Comment:
probably is a lightning library issue but sometimes it does not return it. We need to keep this

Reply:
I just ran our test suite and for 17 payments added to the `lnpayments` collection they all never return a value for `request`.

But yes the library is inconsistent because it also doesn't return a `pending` array as defined in its documentation [here](https://github.com/alexbosworth/lightning/blob/e948ebd7fca54a04b2b0d3715dd3bc9063aef57f/lnd_methods/offchain/get_payment.js#L20) when status is `pending`.

We would like to persist the payment request somehow though, so instead of including it separately I chose to overwrite the place we would expect to see it from the library return if that value is undefined. We can change this behaviour to something else though if this isn't appropriate for here.

Note: there's a similar thing we do with the `paymentHash` property since for pending transactions lnd doesn't give us any info at all even though there should be a `pending` property.

```ts
// Object returned from lib
{
  failed: undefined,
  is_confirmed: false,
  is_failed: false,
  is_pending: true,
  payment: undefined
}
```

Reply:
Just got confirmation, it looks like `request` is never returned for the `getPayment` method: https://github.com/alexbosworth/lightning/issues/50#issuecomment-965593923